diff -uprN emacs-25.1.org/src/keyboard.c emacs-25.1/src/keyboard.c
--- emacs-25.1.org/src/keyboard.c	2016-08-02 17:06:01.000000000 +0900
+++ emacs-25.1/src/keyboard.c	2017-04-17 00:47:59.000000000 +0900
@@ -2467,10 +2467,8 @@ read_char (int commandflag, Lisp_Object 
 	swallow_events (false);		/* May clear input_pending.  */
 
       /* Redisplay if no pending input.  */
-      while (!(input_pending
-	       && (input_was_pending || !redisplay_dont_pause)))
+      while (!input_pending)
 	{
-	  input_was_pending = input_pending;
 	  if (help_echo_showing_p && !EQ (selected_window, minibuf_window))
 	    redisplay_preserve_echo_area (5);
 	  else
